src/k8s/pkg/client/snapd/snap_info.go

```diff
@@ -0,0 +1,31 @@
+package snapd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+type snapdSnapInfoResponse struct {
+	StatusCode int `json:"status-code"`
+}
+
+func (c *Client) GetSnapInfo(snap string) (*snapdSnapInfoResponse, error) {
+	resp, err := c.client.Get(fmt.Sprintf("http://localhost/v2/snaps/%s", snap))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get snapd snap info: %w", err)
+	}
+	defer resp.Body.Close()
+
+	resBody, err := io.ReadAll(resp.Body)
```

Comment:
Maybe we can used buffered read here?
p.s. The response is probably never that big so feel free to ignore this. Just as a precaution.

Reply:
I don't think that's necessary for a few kilobytes (if even that).